### DISCUSSION TITLE
Is it possible to make chunk server to utilize multiple NIC just like multipath does?

### DISCUSSION THREAD

**Discussion by xinminglai** [moosefs/moosefs#320] [Q&A]
## Have you read through available documentation and open Github issues?

Yes

## Is this a BUG report, FEATURE request, or a QUESTION? Who is the indended audience?

QUESTION

#### Your moosefs version and its origin (moosefs.com, packaged by distro, built from source, ...).

3.0.100   personal gentoo ebuild created myself.

**Comment by njaard** [comment]
I don't know if this is relevant to you, but I have point-to-point 10GigE connections between some of my chunkservers. My trick is to create it with the same ip address as the primary port and create a unicast route:

`ip route add unicast LOCAL_ADDRESS proto static src REMOTE_ADDRESS dev DEVICENAME` (on both ends)

Linux will then choose this in preference to normal port because it gives preference to more specific routes. Net result is that connections between the two machines will use that point-to-point 10GigE device, totally transparently and for all applications.

With that said, I've just barely been able to saturate a 10GigE with moosefs.

**Comment by rafamiga** [comment]
Njaard, do you create multiple unicast routes for different interfaces? I can't comprehend the reasoning behind this type of network setup. Would you be so kind as to explain why do you do this?

**Comment by njaard** [comment]
It means that between any two servers that have a dedicated interface, I get the full bandwidth of an entire interface. In a normal (switched) topology, a server shares its bandwidth between all other servers that it communicates through the same interface.

Also note that by "point to point", there's no switch in between, it's a single ethernet cable connecting two servers "crossover style" .

By making a unicast route, I'm just able to make it transparent on an application level.
